scripts/db_migrator.py

```diff
@@ -1024,8 +1026,8 @@ def version_4_0_2(self):
         Version 4_0_2.
         """
         log.log_info('Handling version_4_0_2')
-
-        if self.stateDB.keys(self.stateDB.STATE_DB, "FAST_REBOOT|system"):
+        enable_state = self.stateDB.get(self.stateDB.STATE_DB, 'FAST_RESTART_ENABLE_TABLE|system', 'enable')
+        if self.stateDB.keys(self.stateDB.STATE_DB, "FAST_REBOOT|system") or enable_state == 'true':
```

Comment:
Why is this change needed? The check here was added (https://github.com/sonic-net/sonic-utilities/pull/2839) for old systems (like 201911) where new flag was not present. The newer branches (where `FAST_RESTART_ENABLE_TABLE` flag is present use below changes to delay flex counters:
https://github.com/sonic-net/sonic-swss/pull/1877
https://github.com/sonic-net/sonic-utilities/pull/1768

@shlomibitton can you please review if the above analysis is correct?

Reply:
Hi @vaibhavhd , this issue is only reproduced from 202205 to 202211. In 202205, there is already a db migrator function version_3_0_5 which changes old flag to new flag. However, in 202211, the same logic is in version_4_0_0. While upgrading from 202205 to 202211, 202205 sets new flag and we have a dump.rdb; in 202211 db_migrator just sets flag to false.

Reply:
Hi @stepanblyschak and @arfeigin , could you please provide your comment here?

Reply:
@Junchao-Mellanox correct, good observation.
The suggested change for version_4_0_0 is in place and hence also this change in version_4_0_2 is needed to tie it together so it will all work as expected.

Reply:
@vaibhavhd As wrote since in version_4_0_0 in the db_migrator according to this PR we are adding new state-db entry for fast-reboot instead of the old version it will be more coherent to check for the newer version when migrating to even newer version (4_0_2) and ignore the former state-db entry.

Reply:
I still don't understand why these two lines need to be changed. I understand the change in 4_0_0, but w/o the change in 4_0_2, what is the issue?

I am not saying that this is doing any damage, just that I don't understand why we are fixing this if this is not needed.

Reply:
Hi @vaibhavhd, you are right, this change is not necessary. I have updated code, could you please review again?